src/dtclient/endpoints.go

```diff
@@ -25,11 +25,19 @@ func (dtc *dynatraceClient) getAgentVersionsUrl(os, installerType, flavor, arch
 }
 
 func (dtc *dynatraceClient) getOneAgentConnectionInfoUrl() string {
-	return fmt.Sprintf("%s/v1/deployment/installer/agent/connectioninfo", dtc.url)
+	if dtc.networkZone != "" {
+		return fmt.Sprintf("%s/v1/deployment/installer/agent/connectioninfo?networkZone=%s&defaultZoneFallback=true", dtc.url, dtc.networkZone)
+	} else {
```

Comment:
No need for the else here imho you could just return

Reply:
Fixed!